### PR TITLE
Note string optimizer name is case-insensitive

### PR DESCRIPTION
**Pull request by dunnkers**
A minor documentation addition. For clarity, it can be noted that the optimizer string can be put in any casing in `model.compile`, i.e. case-insensitive. e.g.

```python
model.compile(optimizer='RMSprop')
model.compile(optimizer='rmsprop')
model.compile(optimizer='RMSPROP')
```

→ Are all the same. It makes sense to tell the user they do not have to mind the casing.

This is because whatever the user inputs as an optimizer string is converted to a lowercase string in [optimizers.py#L81](https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/keras/optimizers.py#L81); and then compared to other lower-cased optimizer strings.

Have a great day still ☀️

### CODE REVIEW

**Comment by chenmoneygithub** [comment]
Good catch and thanks for the PR! 

In fact this case thing applies not only to optimizer, but also loss and metrics. We decided that this won't affect users in a significant way, so we will close the PR. Thanks again!
